### PR TITLE
[IMP] web_tour: tour_interactive creation

### PR DESCRIPTION
**Pull request by bastien-ipb**
Before, the manual tour and auto tour were using the same mecanism. Now, it separate to be more appropriate.
The auto tour should keep using a macro engine because it's a automatically executing engine. But the manual tour is a functionality that is interacting with the user. So the use an interactive tour engine is more appropriate.

With this new engine, a new functionality has been added. The tour can now backward. It's used here for backward when the pointer lose the trigger of the step. It will backward until a step that have a trigger on the view.

Task-id: 3922501

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web_tour/static/src/tour_service/tour_interactive.js [hunk 1]
```diff
@@ -0,0 +1,395 @@
...
+    getConsumeEventType(element, runCommand) {
+        const consumeEvents = [];
+        if (runCommand === "click") {
+            consumeEvents.push({
+                name: "click",
+                target: element,
+            });
+
+            // Use the hotkey should also consume
+            if (element.hasAttribute("data-hotkey")) {
```
Comment (bastien-ipb) [inline addons/web_tour/static/src/tour_service/tour_interactive.js:278]: Hotkey does click

**Comment by bastien-ipb** [comment]
Hi @odoo/rd-security , could I have an override ? This is just moved code to another file :)

**Comment by kebeclibre** [comment]
very nice work. Thanks !
robodoo r+

File: addons/web_tour/static/src/tour_service/tour_service.js [hunk 2]
```diff
@@ -243,54 +245,47 @@ export const tourService = {
...
+            let message;
+            if (typeof rainbowManMessage === "function") {
+                message = rainbowManMessage({
+                    isTourConsumed: (name) => consumedTours.has(name),
+                });
+            } else if (typeof rainbowManMessage === "string") {
+                message = rainbowManMessage;
+            } else {
+                message = markup(
+                    _t("<strong><b>Good job!</b> You went through all steps of this tour.</strong>")
```
Comment (pipu-odoo) [inline addons/web_tour/static/src/tour_service/tour_service.js:266]: @odoo/rd-security, markup of static html => no risks

**Comment by kebeclibre** [comment]
robodoo r+

**Pull request closed without merging**